### DISCUSSION TITLE
How to wait for the initial synchronization to finish using materialized Postgres?

### DISCUSSION THREAD

**Discussion by frafra** [ClickHouse/ClickHouse#53112] [Q&A]
Hi!
I am replicating a lot of data and different table, thus it takes some time before all the table are created and populated the first time. Since the `create database` operation for materialized Postgres is asynchronous, how can I just wait to finish before running a query on the replicated tables?

**Comment by melvynator** [comment]
Yes indeed. The process is in the background and is actually never finished given that it reads log file in PostgreSQL to add new rows. The best is probably to check data consistency between the two tables. It can be done fairly easily by using a postgres table engine.
